vpr/src/place/initial_placement.cpp

```diff
@@ -76,10 +80,29 @@ static int check_macro_can_be_placed(t_pl_macro pl_macro, int itype, t_pl_loc he
     // Every macro can be placed until proven otherwise
     int macro_can_be_placed = true;
 
+    //Check if macro is constrained
+    bool macro_constrained = is_macro_constrained(pl_macro);
+    PartitionRegion macro_pr;
+
+    if (macro_constrained) {
+        macro_pr = constrained_macro_locs(pl_macro);
+    }
+
     // Check whether all the members can be placed
     for (size_t imember = 0; imember < pl_macro.members.size(); imember++) {
         t_pl_loc member_pos = head_pos + pl_macro.members[imember].offset;
 
+        //if the macro is constrained, check if the member position is within the PartitionRegion for the macro
```

Comment:
check the floorplan constraint outside the loop, and only for the head member of the macro